src/http-client/node-http2-client.ts

```diff
@@ -149,7 +140,8 @@ class SessionWrapper {
     data: requestData,
     headers: requestHeaders,
     method,
-  }: SessionRequestOptions): Promise<HTTPResponse> {
+    client_timeout_ms,
```

Comment:
let's make life easy for ourselves and use the buffer here.

So the buffer is always controlled in one palce.

Reply:
alternative is to infer the client timeout from defaults in the client.ts and never pass undefined here. and have the buffer be a client only concern. that might be cleaner